### PR TITLE
Don't place shipyard or boats at very small lakes

### PR DESCRIPTION
**Pull request by DjWarmonger**
Fixes cases like this: https://github.com/vcmi/vcmi/issues/939

### CODE REVIEW

File: lib/rmg/WaterProxy.cpp [hunk 1]
```diff
@@ -142,6 +142,13 @@ RouteInfo WaterProxy::waterRoute(Zone & dst)
 				dst.areaPossible().subtract(lake.neighbourZones[dst.getId()]);
 				continue;
 			}
+
+			//Don't place shipyard or boats on the very small lake
```
Comment (nullkiller) [inline lib/rmg/WaterProxy.cpp:146]: Will it try to put any treasures there?

Comment (DjWarmonger) [reply lib/rmg/WaterProxy.cpp:146]: Yep, but it's irrelevant to the gameplay.

Such small lakes are usually placed right next to land guard between the zones.

File: lib/rmg/WaterProxy.cpp [hunk 2]
```diff
@@ -142,6 +142,13 @@ RouteInfo WaterProxy::waterRoute(Zone & dst)
...
+			if (lake.area.getTiles().size() < 25)
```
Comment (Nordsoft91) [inline lib/rmg/WaterProxy.cpp:147]: Are we sure this won't lead to unplayable maps?

Comment (DjWarmonger) [reply lib/rmg/WaterProxy.cpp:147]: In an edge case it should fall back to Monolith conection, but I'll double check.

Comment (DjWarmonger) [reply lib/rmg/WaterProxy.cpp:147]: Yep, the default route is invalid - so if RMG won't find any other connection, it will just place Monolith pair.

**Pull request merged**